tests/integration-tests/tests/scaling/test_scaling.py

```diff
@@ -355,6 +390,57 @@ def _test_keep_or_replace_suspended_nodes(
     assert_num_instances_in_cluster(cluster_name, region, len(static_nodes))
 
 
+def _test_status_check_replacement(
+    remote_command_executor,
+    scheduler_commands,
+    cluster_name,
+    region,
+    partition,
+    num_static_nodes,
+    num_dynamic_nodes,
+    dynamic_instance_type,
+):
+    """Test nodes failing static check are correctly replaced."""
+    logging.info("Testing that nodes failing static check are correctly replaced")
+    job_id = submit_initial_job(
+        scheduler_commands,
+        "sleep 500",
+        partition,
+        dynamic_instance_type,
+        num_dynamic_nodes,
+        other_options="--no-requeue",
+    )
+    static_nodes, dynamic_nodes = assert_initial_conditions(
+        scheduler_commands, num_static_nodes, num_dynamic_nodes, partition, job_id
+    )
+    # Get network interface name from Head node, assuming Head node and Compute are of the same instance type
+    interface_name = remote_command_executor.run_remote_command(
+        "nmcli device status | grep ether | awk '{print $1}'"
+    ).stdout
+    logging.info("Detaching network interface {} on Compute nodes".format(interface_name))
+    # Submit job that will detach network interface on all static nodes, this will cause EC2 status check to fail
```

Comment:
is this true? it looks like we are only detaching the dynamic one

Reply:
Done